### PR TITLE
Trigger apply changed when mouse up

### PR DESCRIPTION
**Pull request by juliaroldi**
Added a new parameters `applyChangesOnMouseUp`. If this parameter is true, when editing an image the last changes will be applied when mouse up event occurs. `applyChangesOnMouseUp` will be false by default.

### CODE REVIEW

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts [hunk 1]
```diff
@@ -179,6 +180,11 @@ export default class ImageEdit implements EditorPlugin {
                     e.preventDefault();
                 }
             },
+            mouseup: e => {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:183]: we already have mouse up event handler at line 185

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/editInfoUtils/applyChange.ts [hunk 2]
```diff
@@ -57,6 +57,10 @@ export default function applyChange(
             newSrc,
         });
         newSrc = event.newSrc;
+    } else {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/editInfoUtils/applyChange.ts:60]: check the new parameter as well.

File: packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts (hunk 1 above)
Comment (juliaroldi) [reply packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:183]: The handler in the switch case is not triggered when the mouse up event happens in the image wrapper.

Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:183]: I see. In that case, we should only attach this event handler when image wrapper is showing, and detach it when hide.

Also please try if we can attach the event to the wrapper element only, but not the whole window.

Comment (juliaroldi) [reply packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:183]: Unless the user always have the cursor on the handles during the whole rotate and resize process, the event will not be triggered. Sometimes the cursor can move out from the handles. 

resizeAndRotate

Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/ImageEdit/ImageEdit.ts:183]: Try pass true as the 3rd parameter when addEventListener, see if that can fix this problem:

```ts
node.addEventListener('mouseup', handler, true /*useCapture*/);
```

If that works, also need to pass true when remove event listener.

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**